iOS/Inventory/DittoManager.swift

```diff
@@ -6,144 +6,242 @@
+                // checking the count of deletions.
+
+                // if the insertions count is greater than zero and others are empty
+                // assume initial load
+                if diff.insertions.count > 0 && diff.deletions.isEmpty
+                    && diff.updates.isEmpty
+                {
+                    let event = SyncEvent.initial
+                    self?.itemsUpdated.send(
+                        (indices: diff.insertions, event: event)
+                    )
+
+                } else {
+                    let event = SyncEvent.update
+                    self?.itemsUpdated.send(
+                        (indices: diff.updates, event: event)
+                    )
                 }
             }
-        )
+        } catch {
+            print("Error: \(error)")
+        }
     }
 
+    /// Prepopulates the "inventories" collection with new items if they do not already exist.
+    ///
+    /// This method executes a DQL query to insert new documents into the "inventories" collection.
+    /// It uses the "INSERT INTO inventories INTIAL DOCUMENTS" statement to ensure that the documents
+    /// are only inserted if they do not already exist, matching the behavior of `.insertDefaultIfAbsent`.
+    ///
+    /// - Parameter itemIds: An array of integers representing the IDs of the items to be inserted.
     func prepopulateItemsIfAbsent(itemIds: [Int]) {
 
-        let allItems = itemIds.map {
-            ["_id": $0,
-             "counter": DittoCounter()]
-        }
-
-        ditto.store.write { transaction in
-            let scope = transaction.scoped(toCollectionNamed: ItemDittoModel.collectionName)
-            allItems.forEach {
-                do {
-                    try scope.upsert($0, writeStrategy: .insertDefaultIfAbsent)
-                } catch {
-                    let dittoErr = (error as? DittoSwiftError)?.errorDescription
-                    assertionFailure(dittoErr ?? error.localizedDescription)
+        // CREATE new items using the INSERT INTO xxx INTIAL statement - must use
+        // initial for first time seeding of documents so sync doesn't screw up
+        // between peers - this matches .insertDefaultIfAbsent in QueryBuilder
+        let query = "INSERT INTO inventories INTIAL DOCUMENTS (:item)"
+
+        Task {
+            // Create a transaction to run inserts into with DQL - this is the equvilant to scoped transaction using store.write
+            // TODO insert doc link to transactions with DQL
+            do {
+                try await ditto.store.transaction { transaction in
+                    for itemId in itemIds {
+                        do {
+                            try await transaction.execute(
+                                query: query,
+                                arguments: [
+                                    "item":
+                                        [
+                                            "_id": itemId,
```

Comment:
For data integrity I think a customer would likely want to insert `counter: 0.0` too here so in the big peer they do not have documents with missing fields; makes it easier to do aggregates later

Reply:
updated